implementation guidance

kazuho opened the pull request:
Closes #25.

marten-seemann commented on draft-ietf-quic-reliable-stream-reset.md:
```diff
@@ -171,6 +171,28 @@ bit for the same stream, the initiator MUST NOT change the Application Error
 
+# Implementation Guidance
+
+Semantically, the CLOSE_STREAM frame is more akin to the FIN bit than to the
+RESET_STREAM frame.
```
Maybe say it can be thought of as a combination of the two?

kazuho replied:
Considering that this is the intro leading to the advice that CLOSE_STREAM should be implemented reusing code to handle FIN, I think we want to say that the frame has more commonality with FIN than RESET_STREAM.

I think the problem might be due to the use of the term"semantically," I've changed that to "In terms of transport machinery" in commit de5fe8b.

PTAL.

marten-seemann commented on draft-ietf-quic-reliable-stream-reset.md:
```diff
@@ -171,6 +171,28 @@ bit for the same stream, the initiator MUST NOT change the Application Error
+
+By sending a CLOSE_STREAM frame, the sender commits to delivering all bytes up
+to the Reliable Size. The state transitions to "Data Sent" on the sender side,
+or to "Size Known" on the receive side. All the bytes up to the Reliable Size
+have to be sent adhering to the flow control.
```
```suggestion
or to "Size Known" on the receive side. Note that the flow control limit might prevent
the sender from sending all bytes up to the Reliable Size at once.
```

marten-seemann commented on draft-ietf-quic-reliable-stream-reset.md:
```diff
@@ -171,6 +171,28 @@ bit for the same stream, the initiator MUST NOT change the Application Error
+bit are:
+- the offset up to which the sender commits to sending might be smaller than
+  Final Size,
+- this offset might get reduced,
```
```suggestion
- this offset might get reduced by subsequent RESET_STREAM_AT frames,
```

marten-seemann commented on draft-ietf-quic-reliable-stream-reset.md:
```diff
@@ -171,6 +171,28 @@ bit for the same stream, the initiator MUST NOT change the Application Error
+  Final Size,
+- this offset might get reduced,
+- the closure is accompanied by an error code, and
+- the CLOSE_STREAM frame never coveys payload like the STREAM frame with the
```
```suggestion
- the CLOSE_STREAM frame don't contain any payload like the STREAM frame with the
```

marten-seemann reviewed (commented):
#27 was merged, so the name of the frame needs to be updated here.

kazuho commented:
@marten-seemann Thank you for the review. I've updated the frame name and adopted all your suggestions modulo the one above, that I changed differently.

marten-seemann commented on draft-ietf-quic-reliable-stream-reset.md:
```diff
@@ -171,6 +171,28 @@ bit for the same stream, the initiator MUST NOT change the Application Error
+  with the FIN bit does.
+
+Therefore, QUIC stacks might implement support for the RESET_STREAM_AT frame by
+extending their code path that deal with the FIN bit.
```
```suggestion
extending their code paths that deal with the FIN bit.
```
